server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1146,7 +1146,9 @@ public synchronized boolean updateIndexMetadata(IndexMetadata indexMetadata) {
      */
     public static boolean same(final Settings left, final Settings right) {
         return left.filter(IndexScopedSettings.INDEX_SETTINGS_KEY_PREDICATE)
-            .equals(right.filter(IndexScopedSettings.INDEX_SETTINGS_KEY_PREDICATE));
+            .equals(right.filter(IndexScopedSettings.INDEX_SETTINGS_KEY_PREDICATE))
+            && left.filter(IndexScopedSettings.ARCHIVED_SETTINGS_KEY_PREDICATE)
+                .equals(right.filter(IndexScopedSettings.ARCHIVED_SETTINGS_KEY_PREDICATE));
```

Comment:
I am thinking why should archive settings impact the equality.

Reply:
`MetadataUpdateSettingsService` will publish the setting with a newer version only when the new settings(after update) and older settings are not equal (i've also added notes in PR description). 
This is the reason why wildcard update(`archived.* = null`) aren't going through and failing silently.